UI problem: ExportToSpreadsheet / WX 3.0 / Centos

On 26 October 2015 at 16:18, Christian Tischer <Christian.Tischer@embl.de> wrote:
Hi Lee,

maybe you saw my post on the cellprofiler-dev mailing list.

i am trying to run the latest CellProfiler from Git on CentoOS.

it almost works, but i sometimes get errors, for instance when clicking on "Select Measurements" in "ExportToSpreadsheet":

Traceback (most recent call last):
  File "/g/almf/software/CellProfiler-2015-10-23/cellprofiler/gui/moduleview.py", line 816, in on_press
    dlg = TreeCheckboxDialog(self.module_panel, d, size=(320,480))
  File "/g/almf/software/CellProfiler-2015-10-23/cellprofiler/gui/treecheckboxdialog.py", line 67, in __init__
    table_sizer.AddGrowableCol(2)
  File "/g/almf/software/anaconda/lib/python2.7/site-packages/wx-3.0-gtk2/wx/_core.py", line 15367, in AddGrowableCol
    return _core_.FlexGridSizer_AddGrowableCol(*args, **kwargs)
PyAssertionError: C++ assertion "!m_cols || idx < (size_t)m_cols" failed at ./src/common/sizer.cpp(1980) in AddGrowableCol(): invalid column index

do you know what this could be?

Tischi

--- a/cellprofiler/gui/moduleview.py
+++ b/cellprofiler/gui/moduleview.py
@@ -813,7 +813,7 @@ class ModuleView:
                     d[None] = leaf_state
                     return leaf_state
                 get_state(d)
-                dlg = TreeCheckboxDialog(self.module_panel, d, size=(320,480))
+                dlg = TreeCheckboxDialog(self.module_panel, d, size=(480,480))
                 dlg.Title = "Select measurements"
                 if dlg.ShowModal() == wx.ID_OK:
                     def collect_state(object_name, prefix, d):

--- a/cellprofiler/gui/treecheckboxdialog.py
+++ b/cellprofiler/gui/treecheckboxdialog.py
@@ -64,7 +64,6 @@ class TreeCheckboxDialog(wx.Dialog):
         self.tree_ctrl.Bind(wx.EVT_LEFT_DOWN, self.on_left_down)
         self.tree_ctrl.Expand(root_id)
         table_sizer = wx.GridBagSizer()
-        table_sizer.AddGrowableCol(2)
         sizer.Add(table_sizer, 0, wx.EXPAND)
         table_sizer.Add(wx.StaticText(self, label='Key:'), (0, 0), flag=wx.LEFT | wx.RIGHT, border=3)
         for i, (bitmap, description) in enumerate((
@@ -75,6 +74,7 @@ class TreeCheckboxDialog(wx.Dialog):
             bitmap_ctrl.SetBitmap(bitmap)
             table_sizer.Add(bitmap_ctrl, (i, 1), flag=wx.RIGHT, border=5)
             table_sizer.Add(wx.StaticText(self, label=description), (i, 2))
+        table_sizer.AddGrowableCol(2)
         sizer.Add(self.CreateStdDialogButtonSizer(wx.CANCEL | wx.OK), 
                   flag=wx.CENTER)
         self.Layout()

